src/helpers/UndoManager/useUndoManager.js

```diff
@@ -26,14 +30,23 @@ const useUndoManager = (
       setCanUndo(nextUndo);
       setCanRedo(nextRedo);
 
+      if (Object.keys(state.blocksClipboard).length !== 0) {
+        const actionType = Object.keys(state.blocksClipboard)[0];
+
+        const blocksData = state.blocksClipboard?.copy
```

Comment:
hmm... I can think of a scenario where the clipboard would be poluted. One thing to know about this "blocks clipboard", it uses localStorage, so it can be used from multiple browser tabs.

Suppose this scenario:

- I select and copy & paste some blocks
- I go on another page on the same site, I make some modifications. 
- If I undo the modifications, is there a chance that the clipboard from the other page would be brought back into action?

Reply:
> If I undo the modifications, is there a chance that the clipboard from the other page would be brought back into action?

AFAIK, it will, since both share the same undo stack and undoo manager saves the state with blocksClipboard and if that object is not empty and doUndo is called irrelevant to the page site it will dispatch an action to setBlocksClipboard.

But I may not be able to understand that do we want this to happen?

Reply:
@MAX-786 Yes, I think mixing the blocks copied from another page with the undo stack of the current page is not desirable. I'm not sure how to solve this issue, in principle we want to know where those blocks come from.

It would be possible (as an idea) if, instead of reading the copied blocks from the store, we would push "block copy actions" into the undo stack. Just an idea, if nothing comes out of it, we may as well leave the current PR as it is. It's not perfect, but it would be fine. @kenmanheimer what do you think of this?

Reply:
Sorry for the delayed response, stuck in clg work. 
> It would be possible (as an idea) if, instead of reading the copied blocks from the store, we would push "block copy actions" into the undo stack.

Yes, indeed this will be the better way of doing it but I wasn't successfully able to accomplish it and I took this simple method to resolve this issue for now and maybe in another issue we can improve it.